src/IssueViz.Security/HotspotsList/TableDataSource/HotspotsTableDataSource.cs

```diff
@@ -97,12 +98,29 @@ public void Add(IAnalysisIssueVisualization hotspot)
                 }
             }
 
-            var hotspotFilePaths = hotspot
-                .GetAllLocations()
-                .Select(x => x.CurrentFilePath)
-                .Distinct(StringComparer.OrdinalIgnoreCase);
+            NotifyHotspotFilesChanged(hotspot);
+        }
 
-            IssuesChanged?.Invoke(this, new IssuesChangedEventArgs(hotspotFilePaths));
+        public void Remove(IAnalysisIssueVisualization hotspot)
+        {
+            var entries = tableEntries.Where(x => x.Identity == hotspot).ToArray();
+
+            if (entries.Length == 0)
+            {
+                return;
+            }
+
+            tableEntries.RemoveAll(x=> x.Identity == hotspot);
```

Comment:
It would be safer to change this condition to `x in entries` rather than repeating the same condition as in line 106. I'd also rename `entries` to `entriesToRemove`.

I guess we're not expecting the array to ever have more than one entry, are we? Assuming that's the case, I'd add an assertion.